File: Common/jetpack/src/main/java/com/ppm/jetpack/MainActivity.java
package com.ppm.jetpack;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.Lifecycle;
import androidx.lifecycle.LifecycleObserver;
import androidx.lifecycle.LifecycleRegistry;
import androidx.lifecycle.OnLifecycleEvent;

public class MainActivity extends AppCompatActivity {

    private IPresenter mPresenter;

    private LifecycleRegistry lifecycleRegistry;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        getLifecycle().addObserver(new MyObserver());
        mPresenter = new MainPresenter();
        getLifecycle().addObserver(mPresenter);

        lifecycleRegistry = new LifecycleRegistry(this);
        lifecycleRegistry.markState(Lifecycle.State.CREATED);
    }

    @Override
    protected void onStart() {
        super.onStart();
        lifecycleRegistry.markState(Lifecycle.State.STARTED);
    }

    @NonNull
    @Override
    public Lifecycle getLifecycle() {
        return lifecycleRegistry;
    }

    @Override
    protected void onResume() {
        super.onResume();
    }

    @Override
    protected void onPause() {
        super.onPause();
    }

    public class MainObserver implements LifecycleObserver {

        @OnLifecycleEvent(Lifecycle.Event.ON_RESUME)
        void onResume() {

        }

        @OnLifecycleEvent(Lifecycle.Event.ON_PAUSE)
        void onPause() {

        }
    }
}

File: Common/kotlindemo/src/main/java/com/ppm/kotlindemo/MainActivity.kt
package com.ppm.kotlindemo

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.alibaba.fastjson.JSONObject
import com.ppm.ppcomon.net.HttpClient
import com.ppm.ppcomon.net.callback.Callback
import com.ppm.ppcomon.utils.LogUtils
import kotlin.collections.set

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val recyclerView: RecyclerView = this.findViewById(R.id.view_list)
        recyclerView.layoutManager = LinearLayoutManager(this)
        val adapter = HisAdapter()
        recyclerView.adapter = adapter

        val dataList: ArrayList<String> = ArrayList()
        dataList.add("1111")
        dataList.add("2222")
        dataList.add("333")
        adapter.mData = dataList
        requestHisList()
    }


    /**
     * 请求历史列表
     */
    private fun requestHisList() {
        val httpClient = HttpClient()
        val params = getCommonParams();
        params["month"] = "1"
        params["day"] = "1"
        val mCallback: Callback = object : Callback {
            override fun onStart() {
            }

            override fun onSuccess(response: JSONObject) {
                LogUtils.d(response.toJSONString())
            }

            override fun onFail(errorCode: Int, errorMessage: String?) {
            }

            override fun onStop() {
            }
        }

        httpClient.sendPostRequest(
            "http://api.juheapi.com/japi/toh",
            params,
            mCallback
        )
    }

    private fun getCommonParams(): MutableMap<String, String> {
        val params = mutableMapOf<String, String>()
        params["key"] = "0dbb338d2f0bf8392d1ff773df23a555"
        params["v"] = "1.0"
        return params
    }
}
